pkg/sink/postgres.go

```diff
@@ -367,26 +344,28 @@ parse:
 }
 
 func (p *PGXSink) performDDL(count int, ddl string) (err error) {
-	ctx := context.Background()
 	if count == 1 {
 		// if the ddl contains only one command tag, than do not run it into begin/commit block
 		// because command like "CREATE INDEX CONCURRENTLY" can't be run inside transaction
-		if _, err = p.conn.Exec(ctx, "commit"); err != nil {
+		if err = p.endPipeline(); err != nil {
 			return err
 		}
+		p.startPipeline()
 	}
-	if _, err = p.conn.Exec(ctx, ddl); err == nil && count == 1 {
-		if _, err = p.conn.Exec(ctx, "begin"); err != nil {
-			return err
-		}
+
+	if err = p.endPipeline(); err != nil {
+		return err
 	}
-	if err == nil {
+	if _, err = p.conn.Exec(context.Background(), ddl); err != nil {
+		return err
+	} else {
 		err = p.schema.RefreshColumnInfo()
+		p.startPipeline()
 	}
 	return
 }
 
-func (p *PGXSink) flushInsert(ctx context.Context) (err error) {
+func (p *PGXSink) flushInsert() (err error) {
```

Comment:
With pipelining, we can probably remove the `flushInsert` baching entirely.

Reply:
But I think batch insert still faster than multiple single insert in a pipeline ? 🤔

Reply:
Additionally, maybe I can flush earlier in a pipeline to allow these DML operations to be executed by PG sooner, rather than waiting until a Sync then flushing these.